### DISCUSSION TITLE
Feature: HLS stream initial bitrate on iOS

### DISCUSSION THREAD

**Discussion by chrstnwhlrt** [TheWidlarzGroup/react-native-video#4955] [Ideas]
### What problem are you trying to solve?

I would like to provide an initial bandwidth for HLS streaming. Currently the Exoplayer in Android uses always the lowest bandwidth available by default and starts increasing the resolution after some chunks.

The initial bitrate issue has been resolved for Android (PR #4480), but iOS still lacks a way to control the starting stream quality.

### Proposed solution

Provide a way to control the initial HLS stream bitrate / starting stream quality on iOS (an initial bitrate estimate, equivalent to what was added for Android in PR #4480).

For reference, the original Android approach: the Exoplayer allows to provide an initial bitrate estimate, which would solve this issue. Somehow a configuration parameter should be passed to the bandwidth meter in:

https://github.com/react-native-video/react-native-video/blob/edf7eee287ebe5fedd34aa1093fec91cdb7f1989/android/src/main/java/com/brentvatne/exoplayer/DefaultReactExoplayerConfig.java

Line 15: Add `setInitialBitrateEstimate(magicParamComingFromReact)` to the bandwidth meter builder param.

### Platforms

iOS

### Additional context

Related Android implementation: PR #4480.

**Comment by mihailpresssport** [comment]
we have encountered this problem on IOS and it is still relevant to us. Video starts playing in terrible quality and improves over time, there must be a way to adjust this

**Comment by vipulbhj** [comment]
Hello, any update on this ? For us, this is happening on ios

**Comment by moskalakamil** [comment]
Hey @chrstnwhlrt, I created PR #4480 where I added this for Android. Unfortunately, on iOS, there's no straightforward way to achieve this, and we would need a workaround. I'll update this issue to focus on iOS only for now, but I don't have any plans to implement this on iOS at the moment so if anyone is interested in working on this, feel free to take it on! 🚀

**Comment by TWG-AI** [comment]
Thanks for the idea — noted for the team to consider.

> I'm an AI assistant, so double-check any specifics I may have mentioned.

**Comment by TWG-AI** [comment]
Quick update: Android initial bitrate support was added in [PR #4480](https://github.com/TheWidlarzGroup/react-native-video/pull/4480) (merged). The iOS side is still open for contributions — AVPlayer doesn't have a direct initial bitrate parameter, so it would need a workaround as noted above.

> I'm an AI assistant — verify current iOS status in the docs.
